### PR TITLE
Capnp

### PR DESCRIPTION
**Pull request by ihsakashi**

### CODE REVIEW

File: packages/capnproto/build.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+TERMUX_PKG_HOMEPAGE=https://capnproto.org/
+TERMUX_PKG_DESCRIPTION="Cap'n Proto tool and library"
```
Comment (xeffyr) [inline packages/capnproto/build.sh:2]: Use normal description here. We all know that package contains cap'n proto stuff, but what is it?
```
Data interchange format and capability-based RPC system.
```

File: packages/capnproto/build.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+TERMUX_PKG_LICENSE="MIT"
+TERMUX_PKG_VERSION=0.8.0
+TERMUX_PKG_SRCURL=https://capnproto.org/capnproto-c++-${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_SHA256=d1f40e47574c65700f0ec98bf66729378efabe3c72bc0cda795037498541c10d
+TERMUX_PKG_BUILD_IN_SRC=true
+
+termux_step_configure() {
+    ./configure \
```
Comment (xeffyr) [inline packages/capnproto/build.sh:10]: Don't run ./configure on your own, especially if you don't know the correct arguments for cross-compilation.

`termux_step_configure` should be overridden only in special cases, e.g. when project doesn't use GNU autotools.

**Review by xeffyr — CHANGES_REQUESTED** [review]

**Comment by ihsakashi** [comment]
Only one thing stopping it building. When makefile runs `test_capnpc_middleman` in the docker, it fails with this:
`./capnp: line 117: /home/builder/.termux-build/capnproto/src/.libs/capnp: cannot execute binary file: Exec format error`.

It works fine building on device. I will submit patch removing test, is that fine?

**Comment by xeffyr** [comment]
> When makefile runs test_capnpc_middleman in the docker, it fails with this

Tests can't be executed. That's known issue and we are just skipping them.

**Comment by ihsakashi** [comment]
So I've removed the tests in Makefile, but i keep getting linker errors when installing. Everytime `test_capnpc_middleman` where the exec error is runs, it outputs dependencies I think and I can't get any further than that. 

https://github.com/capnproto/capnproto/blob/dec294c0de3dda355df11e5b2a69bea78f417516/c%2B%2B/Makefile.am#L429

Can you check it out? I'm stuck :(

**Comment by xeffyr** [comment]
I have switched build.sh to force use cmake as project supports it. Tests seems to be automatically disabled because can't be done when cross-compiling.

**Pull request merged**